external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -191,6 +190,8 @@ pub fn update_compiler_toolchain(
         flavor,
     })?;
     move_table["toolchain-version"] = to_toml_edit_value(&toolchain_version);
+    file.set_len(0)?;
+    file.rewind()?;
```

Comment:
Small fix: we need to truncate the file to erase all previous content when updating these values, now covered by test.